File: backend/API/Domain/Model/Shelter/Pet.cs
using API.Domain.Common.Model;
using API.Domain.Model.Enums;

namespace API.Domain.Model.Shelter
{
    public class Pet : BaseModel
    {
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public string? RescueStory { get; set; }

        public DateOnly? BirthDate { get; set; }

        public decimal? WeightKg { get; set; }

        public string? Slug { get; set; }

        public bool IsVaccinated { get; set; }

        public bool? IsRecommend { get; set; } = false;

        public bool IsSterilized { get; set; }

        public bool IsAdopted { get; set; }
        public int Age { get; set; }

        public PetGender Gender { get; set; }

        public PetSize Size { get; set; }

        public PetStatus Status { get; set; }

        // SPECIES

        public Guid SpeciesId { get; set; }

        public Specie Species { get; set; }
            = null!;

        // MANY TO MANY

        public ICollection<PetBreed> PetBreeds { get; set; } = [];

        public ICollection<PetTrait> PetTraits { get; set; } = [];

        public ICollection<PetVaccine> PetVaccines { get; set; } = [];

        // PHOTOS

        public ICollection<PetPhoto> Photos { get; set; } = [];

        // MEDICAL RECORDS

        public ICollection<MedicalRecord> MedicalRecords { get; set; } = [];

        public ICollection<PetSponsor> PetSponsors { get; set; } = [];
    }
}

File: frontend/src/core/domain/models/shelter/pet.ts
import { PetGenders } from "@/features/system/enums/pet-genders/model/pet-genders.model";
import { PetSizes } from "@/features/system/enums/pet-sizes/model/pet-sizes.model";
import { PetStatus } from "@/features/system/enums/pet-status/model/pet-status.model";
import { Breed } from "./breed";
import { Trait } from "./trait";
import { PetPhotoResponse } from "@/features/shelter/pet-photos/dto/pet-photo-response";
import { VaccineRelation } from "@/features/shelter/vaccine/model/vaccine-relation.model";

export interface Pet {
  id: string;
  name: string;
  description: string;
  rescueStory: string;
  birthDate: Date;
  weightKg: number;
  isSterilized: boolean;
  isRecommend: boolean;
  gender: PetGenders;
  size: PetSizes;
  status: PetStatus;
  speciesName: string;
  isVaccinated: boolean;
  speciesId: string;
  createdAt: Date;
  age: number;
  isAdopted: boolean;

  breeds: Breed[]
  traits: Trait[]
  vaccines: VaccineRelation[]

  photoUrls: PetPhotoResponse[];
}
